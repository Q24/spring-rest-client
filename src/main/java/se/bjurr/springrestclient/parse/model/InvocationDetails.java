package se.bjurr.springrestclient.parse.model;

import java.util.Map;
import java.util.Optional;
import org.springframework.core.ParameterizedTypeReference;
import org.springframework.http.HttpHeaders;
import org.springframework.util.MultiValueMap;

public class InvocationDetails {

  private final RequestDetails requestDetails;
  private final MultiValueMap<String, String> queryParams;
  private final Map<String, String> pathVariables;
  private final Object requestBody;
  private final boolean methodReurnTypeIsResponseEntity;
  private final ParameterizedTypeReference<?> responseType;
  private final HttpHeaders headers;

  public InvocationDetails(
      final RequestDetails requestDetails,
      final MultiValueMap<String, String> queryParams,
      final Map<String, String> pathVariables,
      final Object requestBody,
      final boolean methodReurnTypeIsResponseEntity,
      final ParameterizedTypeReference<?> responseType,
      final HttpHeaders headers) {
    this.requestDetails = requestDetails;
    this.queryParams = queryParams;
    this.pathVariables = pathVariables;
    this.requestBody = requestBody;
    this.methodReurnTypeIsResponseEntity = methodReurnTypeIsResponseEntity;
    this.responseType = responseType;
    this.headers = headers;
  }

  public Map<String, String> getPathVariables() {
    return this.pathVariables;
  }

  public MultiValueMap<String, String> getQueryParams() {
    return this.queryParams;
  }

  public HttpHeaders getHeaders() {
    return this.headers;
  }

  public Optional<Object> findRequestBody() {
    return Optional.ofNullable(this.requestBody);
  }

  public RequestDetails getRequestDetails() {
    return this.requestDetails;
  }

  public ParameterizedTypeReference<?> getResponseType() {
    return this.responseType;
  }

  public boolean isMethodReurnTypeResponseEntity() {
    return this.methodReurnTypeIsResponseEntity;
  }

  @Override
  public String toString() {
    return "InvocationDetails [requestDetails="
        + this.requestDetails
        + ", queryParams="
        + this.queryParams
        + ", pathVariables="
        + this.pathVariables
        + ", requestBody="
        + this.requestBody
        + ", methodReurnTypeIsResponseEntity="
        + this.methodReurnTypeIsResponseEntity
        + ", responseType="
        + this.responseType
        + ", headers="
        + this.headers
        + "]";
  }
}
