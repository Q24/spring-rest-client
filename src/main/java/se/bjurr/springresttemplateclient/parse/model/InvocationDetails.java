package se.bjurr.springresttemplateclient.parse.model;

import java.util.Map;
import java.util.Optional;
import org.springframework.util.MultiValueMap;

public class InvocationDetails {

  private final RequestDetails requestDetails;
  private final MultiValueMap<String, String> queryParams;
  private final Map<String, String> pathVariables;
  private final Object requestBody;
  private final boolean methodReurnTypeIsResponseEntity;
  private final Class<?> responseType;

  public InvocationDetails(
      final RequestDetails requestDetails,
      final MultiValueMap<String, String> queryParams,
      final Map<String, String> pathVariables,
      final Object requestBody,
      final boolean methodReurnTypeIsResponseEntity,
      final Class<?> responseType) {
    this.requestDetails = requestDetails;
    this.queryParams = queryParams;
    this.pathVariables = pathVariables;
    this.requestBody = requestBody;
    this.methodReurnTypeIsResponseEntity = methodReurnTypeIsResponseEntity;
    this.responseType = responseType;
  }

  public Map<String, String> getPathVariables() {
    return this.pathVariables;
  }

  public MultiValueMap<String, String> getQueryParams() {
    return this.queryParams;
  }

  public Optional<Object> findRequestBody() {
    return Optional.ofNullable(this.requestBody);
  }

  public RequestDetails getRequestDetails() {
    return this.requestDetails;
  }

  public Class<?> getResponseType() {
    return this.responseType;
  }

  public boolean isMethodReurnTypeResponseEntity() {
    return this.methodReurnTypeIsResponseEntity;
  }

  @Override
  public String toString() {
    return "InvocationDetails [requestDetails="
        + this.requestDetails
        + ", queryParams="
        + this.queryParams
        + ", pathVariables="
        + this.pathVariables
        + ", requestBody="
        + this.requestBody
        + ", methodReurnTypeIsResponseEntity="
        + this.methodReurnTypeIsResponseEntity
        + ", responseType="
        + this.responseType
        + "]";
  }
}
