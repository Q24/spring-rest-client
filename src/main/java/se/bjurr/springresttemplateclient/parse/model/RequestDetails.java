package se.bjurr.springresttemplateclient.parse.model;

import java.util.Optional;
import org.springframework.http.HttpMethod;
import org.springframework.http.MediaType;

public class RequestDetails {

  private final HttpMethod requestMethod;
  private final String requestPath;
  private final MediaType consumes;
  private final MediaType produces;

  public RequestDetails(
      final HttpMethod requestMethod,
      final String requestPath,
      final MediaType consumes,
      final MediaType produces) {
    this.requestMethod = requestMethod;
    this.requestPath = requestPath;
    this.consumes = consumes;
    this.produces = produces;
  }

  public Optional<MediaType> findConsumes() {
    return Optional.ofNullable(this.consumes);
  }

  public Optional<MediaType> findProduces() {
    return Optional.ofNullable(this.produces);
  }

  public HttpMethod getRequestMethod() {
    return this.requestMethod;
  }

  public String getRequestPath() {
    return this.requestPath;
  }

  @Override
  public String toString() {
    return "RequestDetails [requestMethod="
        + this.requestMethod
        + ", requestPath="
        + this.requestPath
        + ", consumes="
        + this.consumes
        + ", produces="
        + this.produces
        + "]";
  }
}
