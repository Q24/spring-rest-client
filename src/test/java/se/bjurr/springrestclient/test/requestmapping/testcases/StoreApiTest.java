package se.bjurr.springrestclient.test.requestmapping.testcases;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Map;
import org.junit.Test;
import org.springframework.http.MediaType;
import se.bjurr.springrestclient.test.requestmapping.spec.api.StoreApi;
import se.bjurr.springrestclient.test.requestmapping.spec.model.Order;
import se.bjurr.springrestclient.test.utils.BaseApiTest;

public class StoreApiTest extends BaseApiTest<StoreApi> {

  @Override
  public Class<StoreApi> getSutClass() {
    return StoreApi.class;
  }

  @Test
  public void deleteOrder() {
    this.getSut().deleteOrder(123L);
    this.verify();
  }

  @Test
  public void getOrderById() {
    this.mockResponse(MediaType.APPLICATION_JSON_VALUE, "{\"id\":456}");

    final Order actual = this.getSut().getOrderById(456L);

    assertThat(actual.getId()).isEqualTo(456L);
    this.verify();
  }

  @Test
  public void placeOrder() {
    this.mockResponse(
        MediaType.APPLICATION_JSON_VALUE,
        "{\"id\":456,\"petId\":null,\"quantity\":null,\"shipDate\":null,\"status\":null,\"complete\":false}");

    final Order body = new Order().id(123L);
    final Order actual = this.getSut().placeOrder(body);

    assertThat(actual.getId()).isEqualTo(456L);
    this.verify();
  }

  @Test
  public void getInventory() {
    this.mockResponse(MediaType.APPLICATION_JSON_VALUE, "{\"k\":1}");

    final Map<String, Integer> actual = this.getSut().getInventory();

    assertThat(actual).containsEntry("k", 1);
    this.verify();
  }
}
