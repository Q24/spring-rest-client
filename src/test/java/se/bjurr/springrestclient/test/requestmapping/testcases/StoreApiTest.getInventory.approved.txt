[ {
  "url" : "http://localhost:X/store/inventory",
  "method" : "GET",
  "headers" : {
    "Accept" : "application/json",
    "Connection" : "keep-alive"
  },
  "cookies" : { },
  "body" : ""
} ]